pkg/network/so/so_test.go

```diff
@@ -0,0 +1,107 @@
+package so
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func procBashReadPidMapsPathname(t *testing.T, pid int, cmdlineExtra ...string) []string {
+	cmdline := fmt.Sprintf("awk '{print $6}' /proc/%d/maps | sort -u | grep -v '^\\['", pid)
```

Comment:
It's great that we're doing this integration tests here, but from my experience tests that rely too much on the execution environment (having `awk` for example), tend to be too brittle and flaky. Also, these tests are not deterministic and would be very hard to troubleshoot.

So what I would propose is: how about adding a `testutils/procfs` directory tree and passing it as an argument to `Find`?

Reply:
Yeah I got the same feeling to depend too much on classic linux tools (even if it have a good side to test filter vs egrep)
Anyway, I did add procfs directory as kind of an image of /proc for selected list of process
This will allow us to test special case